table/tables/partition.go

```diff
@@ -371,9 +406,57 @@ func (t *partitionedTable) locateRangePartition(ctx sessionctx.Context, pi *mode
 	return idx, nil
 }
 
+func (t *partitionedTable) locateRangePartition(ctx sessionctx.Context, pi *model.PartitionInfo, r []types.Datum) (int, error) {
+	var ret int64
+	if col, ok := t.partitionExpr.Expr.(*expression.Column); ok {
+		ret = r[col.Index].GetInt64()
```

Comment:
Would the column be null?

Reply:
I add some test to cover these cases, PTAL @tiancaiamao